packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -163,6 +167,20 @@ Future<XcodeBuildResult> buildXcodeProject({
   final ProjectMigration migration = ProjectMigration(migrators);
   migration.run();
 
+  final FlutterProject project = FlutterProject.current();
+  if (project.usesSwiftPackageManager && app.project.flutterPluginSwiftPackageManifest.existsSync()) {
+    final SwiftPackageManagerIntegrationMigration spmMigration = SwiftPackageManagerIntegrationMigration(
```

Comment:
Sure, I initially decided against it because it doesn't use any of the functionality of `ProjectMigrator` and `SwiftPackageManagerIntegrationMigration` needs to be async, but your reasoning makes sense. I'll probably make a new PR that converts `ProjectMigrator` to be async